### PR TITLE
optimal assignmnet using exhaustion

### PR DESCRIPTION
**Pull request by ghost**
optimal assignment module updated

### CODE REVIEW

File: optimalAssignment.py [hunk 1]
```diff
@@ -1,4 +1,40 @@
+from itertools import combinations
+from itertools import permutaions
```
Comment (sipah00) [inline optimalAssignment.py]: this should be `permutations`

Comment (sipah00) [reply optimalAssignment.py]: also import `Vector2D` and `dist` function.

File: optimalAssignment.py [hunk 2]
```diff
@@ -1,4 +1,40 @@
...
+taskD --> Dictionary with keys as taks names and values as the position to go ( as an object of the class Vector2D )
+"""
+
 class optimalAssignment(object):
-	def __init__(self, arg):
-		pass
-		
\ No newline at end of file
+	def __init__(self, state, botL, taskD):
+		self.Mat = self.assign( state, botL, taskD)
```
Comment (sipah00) [inline optimalAssignment.py]: remove extra space between `(` and `state`.

**Review by sipah00 — COMMENTED** [review]
Looks good, I have added few points.
Have you run this code on any test case?

File: optimalAssignment.py (hunk 1 above)
Comment (ghost) [reply optimalAssignment.py]: ok

File: optimalAssignment.py (hunk 2 above)
Comment (ghost) [reply optimalAssignment.py]: yup

**Comment by ghost** [comment]
code is yet to be tested on a test case

**Comment by ghost** [comment]
i have made the desired changes

**Comment by ghost** [comment]
additional cases ( minor ) added

**Comment by ghost** [comment]
Added the file to check optimal assignment module against test cases.

**Comment by sipah00** [comment]
@VISHAL-RAJ-RAJ test passed, but why did you write the same code in `opt_test.py`, I think you should just import `optimalAssingment` and make `object` of class, and use it. just change it then it is good to merge.

**Comment by ghost** [comment]
Changed the opt_test.py as desired and made some minor changes.

**Pull request merged**